src/rdb/rdb_raft.c

```diff
@@ -1556,6 +1557,144 @@ rdb_gc_yield(void *arg)
 	return 0;
 }
 
+static void
+rdb_raft_checkpoint_wait(void *arg, uint64_t wait_id, uint64_t *commit_id)
+{
+	struct rdb              *db    = arg;
+	struct rdb_chkpt_record *dcr   = &db->d_chkpt_record;
+	struct umem_store       *store = dcr->dcr_store;
+
+	if (store->stor_ops->so_wal_id_cmp(store, dcr->dcr_commit_id, wait_id) >= 0)
+		goto out;
+
+	ABT_mutex_lock(db->d_raft_mutex);
+	dcr->dcr_waiting = 1;
+	dcr->dcr_wait_id = wait_id;
+	do {
+		sched_cond_wait(db->d_commit_cv, db->d_raft_mutex);
+	} while (store->stor_ops->so_wal_id_cmp(store, dcr->dcr_commit_id, wait_id) < 0);
```

Comment:
You're probably right.  An old habit from using pthread_cond_wait

"Spurious wakeups from the pthread_cond_wait() or pthread_cond_timedwait() functions may occur. Since the return from pthread_cond_wait() or pthread_cond_timedwait() does not imply anything about the value of this predicate, the predicate should be re-evaluated upon such return."

But I don't see that with argobots docs.